src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs

```diff
@@ -323,5 +323,20 @@ public static class WellKnownMemberNames
         /// The required name for the <c>DisposeAsync</c> method used in an await using statement.
         /// </summary>
         public const string DisposeAsyncMethodName = "DisposeAsync";
+
+        /// <summary>
+        /// The required name for the <c>Count</c> property used in a pattern-based Index or Range indexer.
+        /// </summary>
+        public const string CountPropertyName = "Count";
+
+        /// <summary>
+        /// The required name for the <c>Length</c> property used in a pattern-based Index or Range indexer.
+        /// </summary>
+        public const string LengthPropertyName = "Length";
```

Comment:
A bit shocked we haven't had to use `Count` and `Length` yet.

Reply:
Especially given that we lower `foreach` on array and `string` into `for` loops

---
In reply to: [274240382](https://github.com/dotnet/roslyn/pull/34897#discussion_r274240382) [](ancestors = 274240382)

Reply:
In those cases we were looking for specific members on specific types, rather than just any member with that name.